### PR TITLE
added support for firewall rule groups

### PR DESCRIPTION
**Pull request by macnev2013**
## Terraform Tests
- TestAccRoute53ResolverFirewallRuleGroup
  - TestAccRoute53ResolverFirewallRuleGroup_basic
  - TestAccRoute53ResolverFirewallRuleGroup_disappears
  - TestAccRoute53ResolverFirewallRuleGroup_tags
- TestAccRoute53ResolverFirewallRule
  - TestAccRoute53ResolverFirewallRule_basic
  - TestAccRoute53ResolverFirewallRule_block
  - TestAccRoute53ResolverFirewallRule_blockOverride
  - TestAccRoute53ResolverFirewallRule_disappears
- TestAccRoute53ResolverFirewallDomainList
  - TestAccRoute53ResolverFirewallDomainList_basic
  - TestAccRoute53ResolverFirewallDomainList_domains
  - TestAccRoute53ResolverFirewallDomainList_disappears
  - TestAccRoute53ResolverFirewallDomainList_tags

## APIs
**FirewallRuleGroup**
- create_firewall_rule_group
- delete_firewall_rule_group
- get_firewall_rule_group
- list_firewall_rule_groups

**FirewallDomainList**
- create_firewall_domain_list
- delete_firewall_domain_list
- get_firewall_domain_list
- list_firewall_domain_lists

**FirewallDomains**
- update_firewall_domains
- list_firewall_domains

**FirewallRule**
- create_firewall_rule
- delete_firewall_rule
- list_firewall_rules
- update_firewall_rule

## patch
- _matched_arn

## enhancement
- support to generate random hex
- isolated route53resolver from route53

### CODE REVIEW

**Comment by macnev2013** [comment]
@whummer This PR is ready for review.

File: localstack/services/route53resolver/provider.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            raise ResourceNotFoundException(
+                f"Can't find the resource with ID '{firewall_rule_group_id}'. Trace Id: '1-{get_random_hex(8)}-{get_random_hex(24)}'"
+            )
+        return GetFirewallRuleGroupResponse(FirewallRuleGroup=firewall_rule_group)
+
+    def list_firewall_rule_groups(
+        self, context: RequestContext, max_results: MaxResults = None, next_token: NextToken = None
+    ) -> ListFirewallRuleGroupsResponse:
+        region_details = Route53ResolverBackend.get()
+        firewall_group_rules: Optional[FirewallRuleGroupMetadataList] = []
```
Comment (whummer) [inline localstack/services/route53resolver/provider.py:93]: nit: could be changed to:
```
firewall_group_rules: FirewallRuleGroupMetadataList = []
```
(.. or even without the local type hints:)
```
firewall_group_rules = []
```

File: localstack/services/route53resolver/provider.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        for firewall_rule_group in region_details.firewall_rule_groups.values():
+            firewall_group_rules.append(
+                FirewallRuleGroupMetadata(
```
Comment (whummer) [inline localstack/services/route53resolver/provider.py:96]: We have a neat utility `select_from_typed_dict` in `utils.collections` to automatically select only the attributes defined in the model classes:
```
firewall_group_rules.append(select_from_typed_dict(FirewallRuleGroupMetadata, firewall_rule_group))
```

**Review by whummer — APPROVED** [review]
Nice set of changes @macnev2013 , thanks for adding this. 👍  Two smaller nitpick comments, otherwise looks good to me. I understand this is covered by Terraform tests - integration to be discussed in separate PR #6393.

File: localstack/services/route53resolver/provider.py (hunk 1 above)
Comment (macnev2013) [reply localstack/services/route53resolver/provider.py]: Updated.

File: localstack/services/route53resolver/provider.py (hunk 2 above)
Comment (macnev2013) [reply localstack/services/route53resolver/provider.py]: Updated.

**Review by whummer — APPROVED** [review]
Thanks for updating the PR, LGTM! 👍

**Pull request merged**